otherlibs/unix/signals.c

```diff
@@ -97,6 +97,19 @@ CAMLprim value caml_unix_sigsuspend(value vset)
   return Val_unit;
 }
 
+CAMLprim value caml_unix_sigwait(value sigs)
+{
+  sigset_t set;
+  int retcode, signo;
+
+  decode_sigset(sigs, &set);
+  caml_enter_blocking_section();
+  retcode = sigwait(&set, &signo);
```

Comment:
Note: in st_posix.h, the use of this function was conditioned on the HAS_SIGWAIT macro, correesponding to a AC_CHECK_FUNC([sigwait]...) check in the configure. Now it is only conditioned on the POSIX_SIGNALS macro, but the definition of POSIX_SIGNALS only checks for `sigaction` and `sigprocmask`. So now we are (newly) assuming that if those two are available, then `sigwait` also is (and, above, `sigsuspend`).

Reply:
See my helpful commit message in b62ba96d14.  This kind of per-function configure test was appropriate in the 1990's, but nowadays either you're on Windows or you have full POSIX.1-2008 (or at the very least -2001) support.